### PR TITLE
Adding data auth layer.

### PR DESCRIPTION
**Pull request by rossbar**
## What
* Adds Python user interface to authentication layer.

Ports over the example function from @alexanderpg96 for inclusion in the `deepcell-tf` library. This could arguably be included somewhere else (e.g. `deepcell-toolbox`) - I'm open to feedback here. There are still a few things to hammer out on the Python side (tests, default download locations, etc.) as well as with the deepcell hosting app (update URL, etc.)

@msschwartz21 WDYT about the general approach?

### CODE REVIEW

File: deepcell/data/_auth.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    # Extract the filename from the asset_key, which can be a full path
+    fname = os.path.split(asset_key)[-1]
+
+    # Request download URL
+    headers = {"X-Api-Key": access_token}
+    logging.info("Making request to server")
+    resp = requests.post(
+        _api_endpoint, headers=headers, data={"s3_key": asset_key}
+    )
+    resp.raise_for_status()  # Raise exception if not 200 status
```
Comment (rossbar) [inline deepcell/data/_auth.py:66]: Another TODO: Might want to catch `404`'s specifically and raise with the message body which is much more informative than the default HTTPError message.

**Comment by rossbar** [comment]
FTR usage looks something like:

```python
# Import location will be updated, only for testing at this stage
>>> from deepcell.datasets._auth import fetch_data
>>> fetch_data("tissuenet/tissuenet_v1.0.zip")
```

which will download tissuenet to `~/.deepcell/tissuenet_v1.0.zip`[^1]

[^1]: feedback on preferred default download location welcome - where are models/data currently stored by default?

**Comment by msschwartz21** [comment]
Just wanted to drop a link to the source code for the keras function that we have been using in case we want to borrow anything regarding extracting archives or verifying hashes

https://github.com/keras-team/keras/blob/ab566fdc9ae83a88dc4bf149e055773e4d9edf86/keras/utils/data_utils.py#L194

**Review by msschwartz21 — APPROVED** [review]
This looks great to me! I'd like add a step to extract compressed files, but I can do that in the subsequent data PR when I'll also be able to test that functionality at the same time. Thank you!

**Pull request merged**